src/databricks/sql/ae.py

```diff
@@ -83,6 +84,9 @@ def __init__(
         self.query_secret = query_secret
         self.status = status
 
+        if self.status is None:
```

Comment:
I don't love having potentially long running ops in the init.  For one thing, it means one more thing that must be mocked when unit testing, regardless of whether the test needs to interact with status directly.  When I have initialization that is non-trivial, but must be complete for an object to be functional, I tend to put that in a factory method, and try to make the constructor hidden...not sure if we have that capability in python though.  Take this comment with a grain of salt, because depending on the user experience, this point may be outweighed by trying to make the client experience simpler.

Reply:
How about this:

I'll add a new `AsyncExecutionStatus.UNKNOWN` and make that the default.  Then modify the `.status` class member to become a property that accesses a private `._status` member. If `._status==AsyncExecutionStatus.UNKNOWN`, this will fire-off the `poll_for_status` and set it.

This way, the long-running op won't happen until a user actually tries to do something which depends on the status.

Reply:
is there a reason to have it pretend to be a property?  As opposed to just get_status()?  This might be a result of other languages I've programmed in, but in my mind, a property is ideally either a.) a field, or b.) something directly computable from fields.  Making it a function suggests that work will be done to get the value.

Reply:
I have no objection to that. Either way is technically "Pythonic".